contracts/DCAFeeManager/DCAFeeManager.sol

```diff
@@ -1,33 +1,118 @@
 // SPDX-License-Identifier: GPL-2.0-or-later
 pragma solidity >=0.8.7 <0.9.0;
 
+import '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+import '@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol';
+import '@openzeppelin/contracts/utils/Multicall.sol';
 import '../interfaces/IDCAFeeManager.sol';
 import '../utils/Governable.sol';
 
-contract DCAFeeManager is Governable, IDCAFeeManager {
+contract DCAFeeManager is Governable, Multicall, IDCAFeeManager {
+  using SafeERC20 for IERC20;
+
   /// @inheritdoc IDCAFeeManager
   uint16 public constant MAX_TOKEN_TOTAL_SHARE = 10000;
   /// @inheritdoc IDCAFeeManager
   uint32 public constant SWAP_INTERVAL = 1 days;
   /// @inheritdoc IDCAFeeManager
   IDCAHub public immutable hub;
   /// @inheritdoc IDCAFeeManager
+  IWrappedProtocolToken public immutable wToken;
+  /// @inheritdoc IDCAFeeManager
   mapping(address => bool) public hasAccess;
-
-  TargetTokenShare[] internal _distribution;
+  /// @inheritdoc IDCAFeeManager
+  mapping(bytes32 => uint256) public positions; // key(from, to) => position id
```

Comment:
I don't understand how / why this works without overriding. I'm really confused here. It applies to all stuff on the contract.